flaps/flaps_machines.go

```diff
@@ -181,15 +181,22 @@ func (f *Client) GetMany(ctx context.Context, machineIDs []string) ([]*fly.Machi
 	return machines, nil
 }
 
-func (f *Client) List(ctx context.Context, state string) ([]*fly.Machine, error) {
-	getEndpoint := ""
+func (f *Client) List(ctx context.Context, state string, metadata map[string]any) ([]*fly.Machine, error) {
+	ctx = contextWithAction(ctx, machineList)
+
+	var params string
 
 	if state != "" {
-		getEndpoint = fmt.Sprintf("?%s", state)
+		params = fmt.Sprintf("%s&", state)
 	}
 
+	for k, v := range metadata {
+		params += fmt.Sprintf("metadata.%s=%s", k, v)
+	}
```

Comment:
This isn't separating out URL parameters with a `&`. A better approach would be to just use `url.Values` so it can handle encoding too. Also,it doesn't look like `state` is even used by any of the callers of `List()` so you could just remove it.

```go
q := make(url.Values)
for k, v := range metadata {
	q.Set("metadata." + k, v)
}

var getEndpoint = ""
if len(q) > 0 {
	getEndpoint = "?" + q.Encode()
}
```